design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,394 @@
+`--include-namespaced-resources` and `--exclude-namespaced-resources` are used to filter namespace scope resources included or excluded in backup per resource type.
+Restore and other code pieces also use resource filtering will be handled in future releases.
+
+### Parameters Rules
+
+* `--include-namespaced-resources` valid value is comma separated string. Not setting the parameter means including all namespace resources.
+
+* `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources` and `--exclude-namespaced-resources` valid value include `*` and comma separated string.
+
+* `--include-cluster-scope-resources`, `--include-namespaced-resources`, `--exclude-cluster-scope-resources` and `--exclude-namespaced-resources` parameters are mutual exclusive with `--include-cluster-resources`, `--include-resources` and `--exclude-resources` parameters. If both sets of parameters are provisioned, validation failure should be returned.
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` can only contain cluster scope resource type names. If namespace scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` specified resources type lists, validation failure should be returned.
+
+* `--include-namespaced-resources` and `--exclude-namespaced-resources` can only contain namespace scope resource type names. If cluster scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-namespaced-resources` and `--exclude-namespaced-resources` specified resources type lists, validation failure should be returned.
+
+* If  `--include-namespaced-resources` is not present, it means all namespace scope resources are included per resource type.
+
+* If `--include-cluster-scope-resources` is not present, it means no additional cluster resource is included per resource type. Cluster resources are related to the namespace scope resources, which means those are returned in the namespace resources' BackupItemAction's result AdditionalItems array, are still included in backup by default. If the parameter is set, listed resources and namespaced resources related cluster resources are included in backup.
+
+* If the backup contains no resource, validation failure should be returned.
+
+### Using scenarios:
+#### no namespaced resources + no cluster resources
+This is not allowed. Backup or restore cannot contain no resource.
+
+#### no namespaced resources + some cluster resources
+The following command means backup no namespaced resources and some cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-scope-resources=StorageClass
```

Comment:
See comment the above comment about the format of the string items, it should be `storageclasses.storage.k8s.io`

Please also update in the following examples.

Reply:
I add the detailed supported parameter format in the above section.